packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -3472,6 +3472,73 @@ describe("typespec-client-generator-core: decorators", () => {
       strictEqual(test2Method.name, "test2");
       deepStrictEqual(test2Method.apiVersions, ["v2"]);
     });
+    it("default latest GA version with preview", async () => {
+      await runner.compile(
+        `
+        @service
+        @versioned(Versions)
+        @server(
+          "{endpoint}",
+          "Testserver endpoint",
+          {
+            endpoint: url,
+          }
+        )
+        namespace Versioning;
+        enum Versions {
+          v2022_10_01_preview: "2022-10-01-preview",
+          v2024_10_01: "2024-10-01",
+        }
+        op test(): void;
+
+        @route("/interface-v2")
+        interface InterfaceV2 {
+          @post
+          @route("/v2")
+          test2(): void;
+        }
+        `
+      );
+      const sdkVersionsEnum = runner.context.sdkPackage.enums[0];
+      strictEqual(sdkVersionsEnum.name, "Versions");
+      strictEqual(sdkVersionsEnum.usage, UsageFlags.ApiVersionEnum);
+      strictEqual(sdkVersionsEnum.values.length, 1);
+      strictEqual(sdkVersionsEnum.values[0].value, "2024-10-01");
+    });
+    it("default latest preview version with GA", async () => {
+      await runner.compile(
```

Comment:
I'm a bit confused. What is the difference between the two test cases? Seems the versions are the same? And I did not find the version set into emitter options.

Reply:
we assign value to the ordering, because we expect users to input versioning in the correct order. In the first one, I inputted a preview version that occurred before the GA version, while in the second test, I put in a preview version that occurs after the GA version.

Since we default the api version to be the last in the list, this changes the default api version from a GA one in the first test, to a preview one in the second test. Therefore in the first test, we only expose a GA version, while in the second test, we expose both the preview and the GA version

Reply:
I see, do we want to add test cases where user explicitly set API version in emitter options?